core/src/io/anuke/mindustry/entities/type/Player.java

```diff
@@ -490,6 +494,10 @@ public void drawBuildRequests(){
     @Override
     public void update(){
 
+        if((unitGroups[waveTeam.ordinal()].size() == 0) && state.rules.limitedRespawns){
```

Comment:
If respawns are reset after each wave and each call to reset(), why is this necessary?

Reply:
Sure